crates/utils/src/utils/validation.rs

```diff
@@ -299,6 +299,36 @@ pub fn check_url_scheme(url: &Option<Url>) -> LemmyResult<()> {
   }
 }
 
+pub fn is_url_blocked(url: &Option<Url>, blocklist: &RegexSet) -> LemmyResult<()> {
+  if let Some(url) = url {
+    if blocklist.is_match(url.as_str()) {
+      Err(LemmyErrorType::BlockedUrl)?
+    }
+  }
+
+  Ok(())
+}
+
+pub fn check_urls_are_valid(urls: &Vec<String>) -> LemmyResult<Vec<String>> {
+  let mut parsed_urls = vec![];
+  for url in urls {
+    let url = match Url::parse(url) {
```

Comment:
Hopping on the rewrite this statement train, this can also be:

```rust
let url = Url::parse(url).or_else(|e|
    if e == ParseError::RelativeUrlWithoutBase {
        Url::parse(&format!("https://{url}"))
    } else {
        Err(e)
    }
)?;
```